### PR TITLE
Convert to TF 0.12. Add tests. Add Codefresh test pipeline

### PR DESCRIPTION
**Pull request by maximmi**
## what
* Port module to Terraform 0.12
* Pin all providers
* Add example for testing
* Add `bats` and `terratest` for the example
* Add Codefresh badge to point to the test pipeline in `terraform-modules` project
* Update README 

## why
* Module currently does not work with 0.12. Much easier syntax
* Better regression control
* Automatically test the example on every commit and pull request
* Provision resources on AWS in the test account and check the outputs for the correct values
* `terraform-modules` project contains pipelines for all terraform modules

## related
* Closes #13 
* Closes #15 
* Closes #11

### CODE REVIEW

File: examples/complete/fixtures.us-west-1.tfvars [hunk 1]
```diff
@@ -0,0 +1,11 @@
+region                                    = "us-east-2"
```
Comment (johncblandii) [inline examples/complete/fixtures.us-west-1.tfvars:1]: #nit filename is `us-west-1` and this is `us-east-2`.

File: examples/vpc-only/fixtures.us-west-1.tfvars [hunk 2]
```diff
@@ -0,0 +1,5 @@
+region             = "us-east-2"
```
Comment (johncblandii) [inline examples/vpc-only/fixtures.us-west-1.tfvars:1]: Same here. `us-west-1` in the filename and `us-east-2` here.

File: examples/complete/fixtures.us-west-1.tfvars [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+namespace                                 = "eg"
+stage                                     = "test"
+name                                      = "vpc_peering_cross_account"
+requester_aws_assume_role_arn             = ""
+requester_region                          = "us-east-2"
+requester_allow_remote_vpc_dns_resolution = true
+accepter_aws_assume_role_arn              = ""
+accepter_region                           = "us-east-2"
+accepter_allow_remote_vpc_dns_resolution  = true
+availability_zones                        = ["us-east-2b"]
```
Comment (johncblandii) [inline examples/complete/fixtures.us-west-1.tfvars:11]: Just noticed, but these look unused in the `main.tf` unless I'm missing something.

Comment (maximmi) [reply examples/complete/fixtures.us-west-1.tfvars]: they are used in `accepter.tf` and `requester.tf`

**Pull request merged**